modules/shared_vpc_access/main.tf

```diff
@@ -20,16 +20,17 @@ data "google_project" "service_project" {
 
 locals {
   gke_shared_vpc_enabled = contains(var.active_apis, "container.googleapis.com")
-  gke_s_account = local.gke_shared_vpc_enabled ? format(
+  gke_s_account = local.gke_shared_vpc_enabled ? [format(
     "service-%s@container-engine-robot.iam.gserviceaccount.com",
     data.google_project.service_project.number,
-  ) : ""
+  )] : []
   dataproc_shared_vpc_enabled = contains(var.active_apis, "dataproc.googleapis.com")
-  dataproc_s_account = local.dataproc_shared_vpc_enabled ? format(
+  dataproc_s_account = local.dataproc_shared_vpc_enabled ? [format(
     "service-%s@dataproc-accounts.iam.gserviceaccount.com",
     data.google_project.service_project.number
-  ) : ""
-  active_api_s_accounts = compact([local.gke_s_account, local.dataproc_s_account])
+  )] : []
+  active_api_s_accounts = flatten([local.gke_s_account, local.dataproc_s_account])
```

Comment:
I think looping on the actual accounts is actually dangerous. I think something like this would be better:

```
locals {
  apis = {
    "container.googleapis.com": format("service-%s@container-engine-robot.iam.gserviceaccount.com", data.google_project.service_project.number
 }
  active_apis = setintersection(keys(local.apis), var.active_apis
  subnetwork_api = setproduct(local.active_apis, var.shared_vpc_subnets)
}
```

Reply:
I ended up having to do `tolist(setproduct(local.active_apis, var.shared_vpc_subnets))` as it was otherwise a set and throwing error `This value does not have any indices`.  In this case I dont think element ordering matters as it becomes a list of elements of type [api,subnet].